src/elem.rs

```diff
@@ -71,25 +85,10 @@ unary_operation!(Sinh, sinh, f64::sinh);
 unary_operation!(Cosh, cosh, f64::cosh);
 unary_operation!(Tanh, tanh, f64::tanh);
 
-pub fn read_3d_image<T>(path: &str) -> Array3<T>
-where
-    T: Mul<Output = T>,
-    T: Add<Output = T>,
-    T: DataElement,
-    u8: AsPrimitive<T>,
-    i8: AsPrimitive<T>,
-    u16: AsPrimitive<T>,
-    i16: AsPrimitive<T>,
-    u32: AsPrimitive<T>,
-    i32: AsPrimitive<T>,
-    u64: AsPrimitive<T>,
-    i64: AsPrimitive<T>,
-    f32: AsPrimitive<T>,
-    f64: AsPrimitive<T>,
```

Comment:
Since you don't use generics anymore, you can probably remove some crates in your `cargo.toml`?